omniscidb/QueryEngine/Execute.cpp

```diff
@@ -2160,11 +2229,15 @@ hdk::ResultSetTable Executor::executeWorkUnitImpl(
         query_mem_desc_owned.reset(new QueryMemoryDescriptor(
             data_mgr_, config_, 0, QueryDescriptionType::Projection, false));
       }
-
-      query_comp_descs_owned.insert(std::make_pair(dt, std::move(query_comp_desc_owned)));
-      query_mem_descs_owned.insert(std::make_pair(dt, std::move(query_mem_desc_owned)));
+      const ExecutorDeviceType compiled_for_dt{query_comp_desc_owned->getDeviceType()};
+      query_comp_descs_owned[compiled_for_dt] = std::move(query_comp_desc_owned);
+      query_mem_descs_owned[compiled_for_dt] = std::move(query_mem_desc_owned);
     }
 
+    const auto exe_policy =
+        getExecutionPolicy(is_agg, query_mem_descs_owned, ra_exe_unit, query_infos, eo);
+    const ExecutorDeviceType fallback_device{
+        exe_policy->hasDevice(co.device_type) ? co.device_type : ExecutorDeviceType::CPU};
```

Comment:
Shouldn't this be always CPU?

Reply:
Right now main has it this way: [can return requested_device_type (i.e., co.device_type)](https://github.com/intel-ai/hdk/blob/6fd1eecf2c2d4932cbb30cbd2ec3995714a6aa27/omniscidb/QueryEngine/Execute.cpp#L2009C1-L2010C1) as [fallback_device](https://github.com/intel-ai/hdk/blob/6fd1eecf2c2d4932cbb30cbd2ec3995714a6aa27/omniscidb/QueryEngine/Execute.cpp#L2113). Is current main wrong?